crates/sparrow-batch/src/batch.rs

```diff
@@ -77,6 +77,23 @@ impl Batch {
         })
     }
 
+    pub fn into_record_batch(self, schema: Arc<Schema>) -> Option<RecordBatch> {
+        self.data.map(|data| {
+            if let Some(fields) = data.data.as_struct_opt() {
+                let mut columns = Vec::with_capacity(3 + fields.num_columns());
+                columns.extend_from_slice(&[data.time, data.subsort, data.key_hash]);
+                columns.extend_from_slice(fields.columns());
+                RecordBatch::try_new(schema, columns).expect("create_batch")
+            } else {
+                RecordBatch::try_new(
+                    schema,
+                    vec![data.time, data.subsort, data.key_hash, data.data],
+                )
+                .expect("create_batch")
+            }
+        })
+    }
```

Comment:
Think we would want to store the `schema` in the `Batch`? Seems like it could be helpful for this case and might as well if we have it in an `Arc`.

Reply:
Maybe? In a lot of places where we have a batch we currently don't already have a schema. Basically, schemas are only useful when we have a batch. I actually think that I'd like to change sources and sinks to operate on `Batch` rather than `RecordBatch` at which point we wouldn't need to get the schema here.

Specifically, consider:

1. ReadParquet (`RecordBatch`)
2. Convert to Batch (`StructArray`, we have a `DataType` but no `SchemaRef`)
3. Run a `Project` (we have a `DataType` but no `SchemaRef`)
...

By operating on `DataType` and `ArrayRef` we allow reading in a primitive (single column) and writing it. Any place we use a `RecordBatch` we *need* to have named columns. Standardizing on `DataType` and `ArrayRef` allow us to operate on primitives as well as we can operate on a struct array. So I think I'd actually like to go the other way. Thoughts?

Reply:
Yeah, agreed on `Batch (structarray)` being passed everywhere. Not having to do "gimme this column" everywhere will hopefully simplify execution (and reduce code, as you say). 

I missed that the only place this `into-record_batch` is called is in the `WriteChannelPipeline`, which is only used for tests. So I think we're on the right track currently.